homeassistant/components/ads/__init__.py

```diff
@@ -118,6 +212,12 @@ def setup(hass, config):
         return False
 
     hass.data[DATA_ADS] = ads
+
+    # load platforms
+    for component, conf_key in PLATFORMS:
+        if conf_key in conf:
+            hass.helpers.discovery.load_platform(component, DOMAIN, conf, conf)
```

Comment:
The last parameter should be the original config.

Please import the discovery helper function as normal. Using `hass.helpers` is legacy and we've removed it from most places already.

```suggestion
            load_platform(hass, component, DOMAIN, conf, config)
```